### ISSUE TITLE
Disabled items in a ComboBox can still be selected with the mouse

### ISSUE THREAD

**Issue by oztune** [adobe/react-spectrum#8116]
### Provide a general summary of the issue here

The `ComboBox` in `react-aria-components` doesn't prevent an item from being selected with the mouse. Note that the item cannot be navigated to using the keyboard, which is the correct behavior. It's just the mouse interaction that's broken. Also note that this issue does not happen in the `ListBox` component when used outside the `ComboBox`

### 🤔 Expected Behavior?

The disabled item should not be clickable, and it should not be allowed to be chosen as the selected item.

### 😯 Current Behavior

In the following gif, taken on [this docs page](https://react-spectrum.adobe.com/react-aria/ComboBox.html#disabled-options) I'm able to click a disabled item and then press enter to commit the selection.

![Image](https://github.com/user-attachments/assets/962c4efc-7349-4c94-9189-674f643c114e)

### 💁 Possible Solution

I haven't looked at the underlying code and haven't found a workaround

### 🔦 Context

This bug is evident in the official documentation, and is not something that's hard to reproduce.

### 🖥️ Steps to Reproduce

See the gif above

### Version

react-aria 1.8.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS 15.3.2 (24D81)

### 🧢 Your Company/Team

3rd party

**Comment by yihuiliao** [comment]
Thanks for reporting this! Definitely seems like a bug. Seems like it broke here https://github.com/adobe/react-spectrum/pull/7561. Will need to investigate more
